src/Ouroboros/Network/Block.hs

```diff
@@ -47,6 +50,11 @@ class StandardHash b => HasHeader b where
 
     blockInvariant :: b -> Bool
 
+-- | When implementing 'HasHeader', use this method to implement the 'measure'
+-- method of the 'Measured' super class.
+blockMeasure :: HasHeader b => b -> BlockMeasure
+blockMeasure b = BlockMeasure (blockSlot b) (blockSlot b) 1
```

Comment:
The point of this function is the following: all the implementers of `HasHeader` have to implement the `Measured` type class. Instead of always writing the following:

```haskell
instance Measured BlockMeasure SomeBlockType where
  measure b = BlockMeasure (blockSlot b) (blockSlot b) 1
```

They can write:

```haskell
instance Measured BlockMeasure SomeBlockType where
  measure = blockMeasure
```

So it is just a convenience function that 'makes sure' the write measure is used.

Reply:
I see what you mean. My point above still stands though... `HasHeader` _implies_ `Measured`... :thinking:

Reply:
True, but you can use a function that uses the `HasHeader` instance to implement its `Measured` super-class instance, as long as you are not using the `Measured` super-class instance in that function, which is the case here.